src/index.tsx

```diff
@@ -282,6 +281,13 @@ function wrap <Props extends {} = {}> (elm: Elm, opts: Options = {}) {
 
       const app = elmModule.init({
         node: consumed,
+        flags: (() => {
+          let flags = {}
+          for (const prop in props) {
```

Comment:
Just because the props are considered "input" to this library and hence out of our control, I think it would be best to stay away from a `for...in` loop. While I highly doubt `react` does or will ever use more complicated objects with enumerable properties on the prototype, it would be safer to use `for (const prop of Object.keys(props)) {` here, because `Object.keys` will not walk the prototype chain of `props`. It's also somewhat more in line with the existing code base.

I think this more imperative style (mutating an object in a loop) was the right choice in this case, it's isolated so it should be easy to track and read going forward. Just making that `Object.keys` change should be good :+1:

Reply:
Cool, I'll change it - as I said I have no experience, so I had no idea this would go through the prototype too!

Initially I wanted to use some object `filter`, but didn't find one unfortunately (at least not without using some other library)

Reply:
Yeah Javascripts standard library is definitely lacking in some area's. Trying to do this in a more declarative style without any external libraries would probably complicate things unnecessarily so I think the way you went about it is the right call